zgw_consumers/models.py

```diff
@@ -36,7 +37,15 @@ class Service(models.Model):
     header_key = models.CharField(_("header key"), max_length=100, blank=True)
     header_value = models.CharField(_("header value"), max_length=255, blank=True)
     oas = models.URLField(
-        _("OAS"), max_length=1000, help_text=_("URL to OAS yaml file")
+        _("OAS url"), max_length=1000, blank=True, help_text=_("URL to OAS yaml file")
+    )
+    oas_file = models.FileField(
+        _("OAS file"),
+        blank=True,
+        null=True,
```

Comment:
`FileField` under the hood is a string-based field, so you shouldn't make those nullable

```suggestion
```